pkg/util/rangeio/rangeio.go

```diff
@@ -134,20 +150,26 @@ var DefaultConfig = Config{
 // ReadRanges only returns [io.EOF] if one of the ranges is beyond the end of
 // the input source.
 func ReadRanges(ctx context.Context, r Reader, ranges []Range) error {
+	// We store our own start time so we can calculate read throughput at the
+	// end.
+	startTime := time.Now()
+	ctx, span := tracer.Start(ctx, "ReadRanges", trace.WithTimestamp(startTime))
```

Comment:
dskit also has a tracing implementation. Did you consider using that? 
It has similar methods (e.g. StartSpanFromContext) but I don't know the differences. Maybe it does something additional to ensure spans get joined into a single trace?